Fixed springboot example.

adeshmukh opened the pull request:
Fix for https://github.com/kubernetes-sigs/kustomize/issues/792

Before this fix, the following failures were observed with the springboot example:
```bash
$ cd examples/springboot

$ kustomize build overlays/staging
Error: couldn't make target for ../../base: No kustomization file found in /Users/adeshmukh/github.com/kubernetes-sigs/kustomize/examples/springboot/base. Kustomize supports the following kustomization files: kustomization.yaml, kustomization.yml, Kustomization

$ kustomize build overlays/production
Error: couldn't make target for ../../base: No kustomization file found in /Users/adeshmukh/github.com/kubernetes-sigs/kustomize/examples/springboot/base. Kustomize supports the following kustomization files: kustomization.yaml, kustomization.yml, Kustomization
```

With this change, both `kustomize build overlays/staging` and `kustomize build overlays/production` work fine.

Liujingfang1 commented on examples/springboot/base/kustomization.yaml:
```diff
@@ -0,0 +1,4 @@
+resources:
+- deployment.yaml
+- service.yaml
```
Please also add a configmapGenerator in the base as
```
configMapGenerator:
- files:
  - application.properties
  name: demo-configmap
```
And in the base directory, add a file `application.properties`
```
app.name=Kustomize Demo
```

adeshmukh-quizlet replied:
Thanks for your review. I will update this PR with the changes tonight.

Liujingfang1 commented on examples/springboot/overlays/production/kustomization.yaml:
```diff
@@ -1,7 +1,8 @@
 bases:
 - ../../base
-patchesStrategicMerge:
+resources:
```
No need to change this file when you add configmapGenerator in the base

Liujingfang1 commented on examples/springboot/overlays/staging/kustomization.yaml:
```diff
@@ -3,9 +3,9 @@ bases:
   literals:
     - foo=bar
   env: staging.env
+
```
No need to change this file when you add configmapGenerator in the base.

monopole commented:
The [README](https://github.com/kubernetes-sigs/kustomize/blob/master/examples/springboot/README.md) text needs to change too.  It's not editing the kustomization file in the base.

The example would be clearer if all the content (the service and deployment yaml) were created directly in the [README](https://github.com/kubernetes-sigs/kustomize/blob/master/examples/springboot/README.md) via the "here" document trick - rather than using curl.

Please removing the curl commands, and the files in the repo, and use this
```
cat <<EOF >$DEMO_HOME/base/whatever.yaml
blah blah
```
to place the content directly into the correct position below `$DEMO_HOME`.

adeshmukh commented:
@monopole Currently the springboot example has:
```
springboot
├── README.md
├── base
│   ├── deployment.yaml
│   └── service.yaml
└── overlays
    ├── production
    │   ├── application.properties
    │   ├── healthcheck_patch.yaml
    │   ├── kustomization.yaml
    │   ├── memorylimit_patch.yaml
    │   └── patch.yaml
    └── staging
        ├── application.properties
        ├── kustomization.yaml
        └── staging.env
```
Are you recommending that this be reduced to:
```
springboot
 └── README.md
```
And have all the other files created via `HEREDOC` in the README?

adeshmukh commented:
FWIW, I notice this note in the README:
> To keep this document shorter, the base resources
> needed to run springboot on a k8s cluster are off in a
> supplemental data directory rather than declared here
> as HERE documents.

monopole commented:
Yeah, i think i wrote that, or wrote the original from which this text was copied.

I think that inlined files, that one creates as one reads, are generally more educational than being presented with a directory full of files as a _fait accompli_.  In the former one builds it up from scratch and cannot help to learn something about the arrangement and the order of creation - at the risk of much longer content.  In the latter there's less content in the markdown, but one must be more of a detective to understand how, why and what order in which the files were made (when that matters).

In the case here, i suppose the example is just too big to keep everything inline.

So, thank's for refactoring / fixing this.  Please do whatever you think is best, it's good to have multiple viewpoints.  Just want it to make sense. 

In your breakdown of 1 vs 2 (layout vs edit commands) i think the layout is more important.   Feel free to remove the edit commands.

We really should have a standalone section describing edit commands, and keep that discussion distinct from build examples.

adeshmukh commented:
@monopole thanks for your thoughtful comments on this PR.

Based on your suggestion, I have made the following changes to the README.md and updated the supporting files to match the description:
1. Removed the references to `kustomize edit ...` commands. Instead the README.md now explains the contents of the springboot example.
2. Retained the aspects of the original example that resulted from the `kustomize edit ...` commands. This required changes to the provided `base` and `overlays` files.

I'm trying to work through the CI failures. I don't quite understand why the `./bin/pre-commit.sh` does not appear to start the README `mdrip` tests at the base of the repository.

monopole commented:
appears to be abandoned. reopen if not
